src/actions/sources/blackbox.js

```diff
@@ -17,16 +17,45 @@ import type { ThunkArgs } from "../types";
 
 export function toggleBlackBox(source: Source) {
   return async ({ dispatch, getState, client, sourceMaps }: ThunkArgs) => {
-    const { isBlackBoxed, id } = source;
+    const { isBlackBoxed } = source;
 
     if (!isBlackBoxed) {
       recordEvent("blackbox");
     }
 
+    const startLocation = await sourceMaps.getGeneratedLocation(
+      { sourceId: source.id, line: 1, column: 0 },
+      source
+    );
+
+    const lines = (await sourceMaps.getOriginalSourceText(source)).text.split(
+      "\n"
+    );
+
+    let endLocation = { line: null };
+    let attempt = 0;
+    while (endLocation.line === null) {
+      const line = lines.length - attempt;
+      const column = lines[line - 1].length - 1;
+      console.log(line, column);
+      endLocation = await sourceMaps.getGeneratedLocation(
```

Comment:
why not map at the end... as in find the last line/col and map that. Lets assume a good world :)

Reply:
I tried mapping the last line and it's last column, but it mapped to nothing. Then I tried to go up the lines to eventually find a mapping. 
This looks bad to me, still trying to find a better way

Reply:
CC @loganfsmyth mind sketching a proof of concept